code/modules/clothing/masks/chemmask.dm

```diff
@@ -0,0 +1,335 @@
+
+/obj/item/clothing/mask/chemmask/proc/tank_volume_check(mob/user) //Alerts the user when the tank runs out of reagents. Does not alert them more than once per emptying, it must be refilled and then run dry again to alert them again.
+	var/obj/item/weapon/reagent_containers/chempack/P = user.back
+	if (P.is_empty() && empty == 0 && tankactive)
+		empty = 1
+		user << "<span class='warning'>The chemical pack is empty!</span>"
+	else if (!P.is_empty())
+		empty = 0
+
+/obj/item/clothing/mask/chemmask/proc/beaker_check(mob/user) //Checks whether there is a beaker in the pack, in order to determine whether to show the beaker-specific verbs.
+	var/obj/item/weapon/reagent_containers/chempack/P = user.back
+	if (P.beaker)
+		hasbeaker = 1
+		verbs += /obj/item/clothing/mask/chemmask/verb/set_beaker_injection_method
+	else
+		hasbeaker = 0
+		verbs -= /obj/item/clothing/mask/chemmask/verb/set_beaker_injection_method
+		verbs -= /obj/item/clothing/mask/chemmask/verb/set_beaker_threshold
+		verbs -= /obj/item/clothing/mask/chemmask/verb/set_beaker_time_interval
+		verbs -= /obj/item/clothing/mask/chemmask/verb/set_beaker_threshold_reagent
+		verbs -= /obj/item/clothing/mask/chemmask/verb/set_beaker_injection_rate
+
+/obj/item/clothing/mask/chemmask/proc/beaker_volume_check(mob/user) //Alerts the user when the auxiliary beaker is empty. Unlike the tank alert, this alert plays a sound, since the auxiliary beaker will most likely have higher-priority chems in it.
+	var/obj/item/weapon/reagent_containers/chempack/P = user.back
+	var/obj/item/weapon/reagent_containers/glass/B = P.beaker
+	if (B.is_empty() && beakerempty == 0)
+		beakerempty = 1
+		playsound(get_turf(src),'sound/mecha/internaldmgalarm.ogg', 100, 1)
+		user << "<span class='warning'>The auxiliary beaker is empty!</span>"
+	else if (!B.is_empty())
+		beakerempty = 0
+
+
+/obj/item/clothing/mask/chemmask/proc/mask_shutdown(mob/user) //Removes most verbs upon toggling the mask off, but not all. The user keeps access to the verbs to toggle connection to the tank and beaker.
+	power = 0
+	icon_state = "chemmask0"
+	user.update_inv_wear_mask()
+	verbs -= /obj/item/clothing/mask/chemmask/verb/set_pack_injection
+	verbs -= /obj/item/clothing/mask/chemmask/verb/set_beaker_injection_method
+	verbs -= /obj/item/clothing/mask/chemmask/verb/set_beaker_threshold
+	verbs -= /obj/item/clothing/mask/chemmask/verb/set_beaker_time_interval
+	verbs -= /obj/item/clothing/mask/chemmask/verb/set_beaker_threshold_reagent
+	verbs -= /obj/item/clothing/mask/chemmask/verb/set_beaker_injection_rate
+
+/obj/item/clothing/mask/chemmask/proc/start_flow(mob/user)
+	spawn()
+		while(power) //Check whether the user has all the parts to the gear, and whether they need to be notified of an empty container.
+			pack_check(user)
+			mask_check(user)
+			tank_volume_check(user)
+			beaker_check(user)
+			sleep(1) //I tried these loops without the sleep(1), it caused the server to lock up when the loop was entered.
+	spawn()

```

Comment:
This entire fucking section is just a load of spawns, stop this.  Use process() or something, this is insane.